mlflow/xgboost.py

```diff
@@ -255,10 +255,32 @@ def log_model(
 
 
 def _load_model(path):
+    """
+    Load Model Implementation.
+
+    :param path: Local filesystem path to the top-level MLflow Model directory.
+    """
     import xgboost as xgb
 
-    model = xgb.Booster()
-    model.load_model(os.path.abspath(path))
+    flavor_conf = _get_flavor_configuration(model_path=path, flavor_name=FLAVOR_NAME)
+
+    if "data" in flavor_conf:
+        # XGBoost Booster models saved in MLflow (<x.x.x) specify
+        # the ``data`` field within its flavor configuration.
+        # In this case, we create a Booster() instance and load model weights.
+        model_class = "Booster"
+        xgb_model_path = os.path.join(path, flavor_conf["data"])
+    else:
+        # In contrast, XGBoost models saved in new MLflow (>=x.x.x) do not
+        # specify the ``data`` field within its flavor configuration.
+        # We use ``model_class`` to specify its XGBoost model class.
+        # In this case, we first get the XGBoost model from
+        # its flavor configuration and then create an instance based on its class.
+        model_class = flavor_conf.get("model_class", "xgboost.core.Booster").split(".")[-1]
+        xgb_model_path = os.path.join(path, "model.xgb")
```

Comment:
```suggestion
    model_dir = os.path.dirname(path)
    flavor_conf = _get_flavor_configuration(model_path=model_dir, flavor_name=FLAVOR_NAME)
    model_class = flavor_conf.get("model_class", "xgboost.core.Booster").split(".")[-1]
        xgb_model_path = os.path.join(path, "model.xgb")
```

Can we use `os.path.dirname` to get the directory that stores model artifacts?

Reply:
Can we load a model class using `importlib`? 

```python
>>> model_class = "xgboost.core.Booster"
>>> module, cls = model_class.rsplit(".", maxsplit=1)
>>> getattr(importlib.import_module(module), cls)
<class 'xgboost.core.Booster'>
```

Reply:
Thanks for the suggestion! I will update them in the next commit shortly.

Reply:
> Can we use `os.path.dirname` to get the directory that stores model artifacts, and pass the directory to `_get_flavor_configuration`?

In the case where `path` is a folder not a model file, it could lead to its parent directory, causing loading error. But I got your idea. Trying to figure out a solution. Thanks for your suggestion!